runtime-modules/content/src/tests/creator_tokens/finalize_creator_token_sale.rs

```diff
@@ -213,3 +215,47 @@ fn successful_finalize_curator_channel_creator_token_sale_by_lead() {
         );
     })
 }
+
+#[test]
+fn purchase_tokens_on_sale_fails_on_frozen_pallet() {
+    with_default_mock_builder(|| {
+        ContentTest::with_member_channel().setup();
+        IssueCreatorTokenFixture::default().call_and_assert(Ok(()));
+        InitCreatorTokenSaleFixture::default().call_and_assert(Ok(()));
+
+        let freeze_result = Token::set_frozen_status(RawOrigin::Root.into(), true);
+        assert_ok!(freeze_result);
+
+        let amount = DEFAULT_CREATOR_TOKEN_ISSUANCE - 1;
+        let existential_deposit: u64 = <Test as balances::Config>::ExistentialDeposit::get().into();
+        increase_account_balance_helper(
+            SECOND_MEMBER_ACCOUNT_ID,
+            existential_deposit + DEFAULT_CREATOR_TOKEN_SALE_UNIT_PRICE * amount,
+        );
+
+        let result = project_token::Module::<Test>::purchase_tokens_on_sale(
+            RuntimeOrigin::signed(SECOND_MEMBER_ACCOUNT_ID),
+            project_token::Module::<Test>::next_token_id() - 1,
+            SECOND_MEMBER_ID,
+            amount,
+        );
+        assert_noop!(result, project_token::Error::<Test>::PalletFrozen);
```

Comment:
The use of `assert_noop!` here is equivalent to `assert_err!`
If you really want to assert no storage was modified the expression returning the result should be the argument to the macro like so:

```suggestion
        assert_noop!(
          project_token::Module::<Test>::purchase_tokens_on_sale(
            RuntimeOrigin::signed(SECOND_MEMBER_ACCOUNT_ID),
            project_token::Module::<Test>::next_token_id() - 1,
            SECOND_MEMBER_ID,
            amount),
          project_token::Error::<Test>::PalletFrozen
        );
```